lib/asciidoctor/js/opal_ext/file.rb

```diff
@@ -76,25 +76,87 @@ def readlines
     File.readlines(@path)
   end
 
-  def self.readlines(path, separator = $/)
-    content = File.read(path)
-    content.split(separator)
-  end
+  class << self
+
+    def read(path)
+      case JAVASCRIPT_IO_MODULE
+      when 'node'
+        %x(return require('fs').readFileSync(path, 'utf8'))
+      when 'phantomjs'
+        %x(return require('fs').read(path))
+      when 'java_nio'
+        %x(
+          var Paths = Java.type('java.nio.file.Paths');
+          var Files = Java.type('java.nio.file.Files');
+          var lines = Files.readAllLines(Paths.get(path), Java.type('java.nio.charset.StandardCharsets').UTF_8);
+          var data = [];
+          lines.forEach(function(line) { data.push(line); });
+          return data.join("\n");
+        )
+      when 'xmlhttprequest'
+        %x(
+          var data = '';
+          var status = -1;
+          try {
+            var xhr = new XMLHttpRequest();
+            xhr.open('GET', path, false);
+            xhr.addEventListener('load', function() {
+              status = this.status;
+              // status is 0 for local file mode (i.e., file://)
+              if (status == 0 || status == 200) {
+                data = this.responseText;
+              }
+            });
+            xhr.overrideMimeType('text/plain');
+            xhr.send();
+          }
+          catch (e) {
+            status = 0;
+          }
+          // assume that no data in local file mode means it doesn't exist
+          if (status == 404 || (status == 0 && data == '')) {
+            throw #{IOError.new `'No such file or directory: ' + path`};
+          }
+          return data;
+        )
+      when 'spidermonkey'
+        %x(return read(path))
+      else
+        ''
+      end
+    end
 
-  # TODO use XMLHttpRequest HEAD request unless in local file mode
-  def self.file?(path)
-    true
-  end
+    def readlines(path, separator = $/)
+      content = File.read(path)
+      content.split(separator)
+    end
 
-  def self.readable?(path)
-    true
-  end
+    # TODO use XMLHttpRequest HEAD request unless in local file mode
+    def file?(path)
+      true
+    end
 
-  def self.read(path)
-    # REMIND will be overriden by a specific implementation
-    ''
-  end
+    def readable?(path)
+      true
+    end
+
+    def read(path)
+      # REMIND will be overriden by a specific implementation
+      ''
+    end
 
+    alias :super_expand_path :expand_path
+
+    # Monkey-patch to keep path unchanged on absolute path:
+    # * URLs starting with file://, http:// or https://
+    def expand_path(path, basedir = nil)
```

Comment:
I don't think we need this override anymore now that we have taken a different approach.

Reply:
I think you're right, I will try to remove it.